packages/cli/src/environments/sourceControl/sourceControlExport.service.ee.ts

```diff
@@ -239,23 +239,28 @@ export class SourceControlExportService {
 				);
 			}
 			await Promise.all(
-				credentialsToBeExported.map(async (sharedCredential) => {
-					const { name, type, nodesAccess, data, id } = sharedCredential.credentials;
-					const credentialObject = new Credentials({ id, name }, type, nodesAccess, data);
-					const plainData = credentialObject.getData();
-					const sanitizedData = this.replaceCredentialData(plainData);
-					const fileName = this.getCredentialsPath(sharedCredential.credentials.id);
-					const sanitizedCredential: ExportableCredential = {
-						id: sharedCredential.credentials.id,
-						name: sharedCredential.credentials.name,
-						type: sharedCredential.credentials.type,
-						data: sanitizedData,
-						nodesAccess: sharedCredential.credentials.nodesAccess,
-					};
-					this.logger.debug(`Writing credential ${sharedCredential.credentials.id} to ${fileName}`);
-					return await fsWriteFile(fileName, JSON.stringify(sanitizedCredential, null, 2));
-				}),
+				credentialsToBeExported
+					.filter((sharing) => sharing.role === 'credential:owner')
```

Comment:
No filtering before, so this file was being re-written as many times as there were sharings, instead of only once per `credential:owner` sharing.

Reply:
Ah good finding!

Reply:
Can we perhaps add this as a `where` filter to the query above?

Reply:
Addressed at [e157ebd](https://github.com/n8n-io/n8n/pull/8955/commits/e157ebd5149f61c484d06215a3e24023e2d59b36)